firebase-perf/src/iosMain/kotlin/dev/gitlive/firebase/perf/metrics/Trace.kt

```diff
@@ -0,0 +1,14 @@
+package dev.gitlive.firebase.perf.metrics
+
+import cocoapods.FirebasePerformance.FIRTrace
+
+actual class Trace(val ios: FIRTrace?) {
```

Comment:
optional?

Reply:
Also probably should be an internal constructor?

Reply:
The FIRTrace returned by traceWithName is nullable. If we wanted to make sure we had a trace we could do something like this in the `Performance` class

```
actual fun newTrace(traceName: String): Trace {

        val trace = FIRPerformance.sharedInstance().traceWithName(traceName)

        checkNotNull(trace)

        return Trace(trace)
    }
```

Reply:
It's interesting that the iOS return value is nullable but the Android one is non-null (from what I can tell). [This issue](https://github.com/firebase/firebase-ios-sdk/issues/8583) speaks to some of the nuances of why it might have to be nullable on iOS, though I don't know why those wouldn't apply to Android (maybe Android throws an Exception?) - empty string, string > 100 chars.

Might be worth `assert()`ing here so that it crashes in dev but not prod

Reply:
this is a wrapper library so it probably should mimic what the native library does. If it returns optional, then it should deal with it being optional.